fish-rust/src/wutil/wrealpath.rs

```diff
@@ -28,17 +31,16 @@ pub fn wrealpath(pathname: &wstr) -> Option<WString> {
     // outside of WSL + Cygwin (which should be fairly Unix-like anyways)
     let narrow_res = canonicalize(OsStr::from_bytes(&narrow_path));
 
-    if let Ok(result) = narrow_res {
-        real_path.extend_from_slice(result.as_os_str().as_bytes());
+    let real_path = if let Ok(result) = narrow_res {
+        result.into_os_string().into_vec()
     } else {
         // Check if everything up to the last path component is valid.
         let pathsep_idx = narrow_path.iter().rposition(|&c| c == b'/');
 
         if let Some(0) = pathsep_idx {
```

Comment:
Can you use a regular comparison here? (`if pathsep_idx == Some(0) {`)
There's no need to use let binding assignment.

Reply:
Ah yes, good point. I got so used to doing it for any assignment that I forgot we didn't always need it.